sunbeam-python/sunbeam/plugins/interface/v1/base.py

```diff
@@ -0,0 +1,310 @@
+
+        @click.command()
+        @click.option(
+            "-t",
+            "--token",
+            help="Ubuntu Pro token to use for subscription attachment",
+            prompt=True,
+        )
+        def enable_subcmd(self, token: str):
+            pass
+
+        The user can invoke the above commands like:
+
+        sunbeam enable subcmd
+        sunbeam disable subcmd
+        sunbeam troubleshoot subcmd
+        """
+
+    def register(self, cli: click.Group):
+        """Register plugin groups and commands."""
+        LOG.debug(f"Registering plugin {self.name}")
+        if not self.validate_commands():
+            LOG.warning(f"Not able to register the plugin {self.name}")
+            return
+
+        groups = utils.get_all_registered_groups(cli)
+        LOG.debug(f"Registered groups: {groups}")
+        for group, commands in self.commands().items():
+            group_obj = groups.get(group)
+            if not group_obj:
+                cmd_names = [command.get("name") for command in commands]
+                LOG.warning(
+                    f"Plugin {self.name}: Not able to register command "
+                    f"{cmd_names} in group {group} as group does not exist"
+                )
+                continue
+
+            for command in commands:
+                cmd = command.get("command")
+                cmd_name = command.get("name")
+                if cmd_name in group_obj.list_commands({}):
+                    if isinstance(cmd, click.Command):
+                        LOG.warning(
+                            f"Plugin {self.name}: Discarding adding command "
+                            f"{cmd_name} as it already exists in group {group}"
+                        )
+                    else:
+                        # Should be sub group and already exists
+                        LOG.debug(
+                            f"Plugin {self.name}: Group {cmd_name} already "
+                            f"part of parent group {group}"
+                        )
+                    continue
+
+                cmd.callback = ClickInstantiator(cmd.callback, type(self))
+                group_obj.add_command(cmd, cmd_name)
+                LOG.debug(
+                    f"Plugin {self.name}: Command {cmd_name} registered in "
+                    f"group {group}"
+                )
+
+                # Add newly created click groups to the registered groups so that
+                # commands within the plugin can be registered on group.
+                # This allows plugin to create new groups and commands in single place.
+                if isinstance(cmd, click.Group):
+                    groups[cmd_name] = cmd
+
+
+class EnableDisablePlugin(BasePlugin):
+    interface_version = Version("0.0.1")
+
+    def __init__(self, name: str):
+        super().__init__(name=name)
+
+    @property
+    def enabled(self) -> bool:
+        info = self.get_plugin_info(self.plugin_key)
```

Comment:
```suggestion
        info = self.get_plugin_info()
```

get_plugin_info does not take an argument, is that expected ?